### PR TITLE
[Rgen] Simplify the way we execute tests on all platforms.

### PR DESCRIPTION
**Pull request by mandel-macaque**
There is no need to write a line per platform when we can use an attribvute for that.

### CODE REVIEW

**Review by dalexsoto — APPROVED** [review]
Clever!

File: tests/common/ConfigurationXUnit.cs [hunk 1]
```diff
@@ -47,11 +47,41 @@ public override IEnumerable<object []> GetData (MethodInfo testMethod)
...
+		}
+
+		public override IEnumerable<object []> GetData (MethodInfo testMethod)
+		{
+			foreach (var platform in Configuration.GetAllPlatforms (true)) {
+				if (!Configuration.IsEnabled (platform))
+					continue;
+				var parameters = dataValues.Prepend (platform).ToArray ();
+				yield return parameters;
+			}
```
Comment (rolfbjarne) [inline tests/common/ConfigurationXUnit.cs:67]: This is even simpler:

```suggestion
			return Configuration.
				GetIncludedPlatforms (true).
				Select (v => dataValues.Prepend (platform).ToArray ());
```

File: tests/common/ConfigurationXUnit.cs [hunk 2]
```diff
@@ -47,11 +47,41 @@ public override IEnumerable<object []> GetData (MethodInfo testMethod)
...
 		}
+
+		public static bool IsEnabled (ApplePlatform platform)
+			=> platform switch {
+				ApplePlatform.iOS => include_ios,
+				ApplePlatform.TVOS => include_tvos,
+				ApplePlatform.MacCatalyst => include_maccatalyst,
+				ApplePlatform.MacOSX => include_mac,
+				_ => false
+			};
```
Comment (rolfbjarne) [inline tests/common/ConfigurationXUnit.cs:85]: And then this is not needed anymore:

```suggestion

```

**Review by rolfbjarne — APPROVED** [review]

File: tests/common/ConfigurationXUnit.cs (hunk 2 above)
Comment (mandel-macaque) [reply tests/common/ConfigurationXUnit.cs:85]: I want to keep this one because it is useful in other use cases.

**Review by stephen-hawley — APPROVED** [review]

**Pull request merged**